projects/ADuCM3029_demo_cn0357/src/cn0357_app.cpp

```diff
@@ -0,0 +1,395 @@
+ DataPacket_t data_pkt = {0x02, 0, 0, 0, 0};
+
+/* Local Bluetooth Functions */
+static void ApplicationCallback(void * pCBParam, uint32_t Event, void * pArg);
+static void SetAdvertisingMode(void);
+static void InitBluetooth(void);
+static uint8_t GasSensorBluetoothMode(Gas *gas);
+#else
+static uint8_t GasSensorStandaloneMode(Gas *gas);
+#endif
+
+/* Local Functions */
+static void InitSystem(void);
+
+/* Defined in pinmux.c */
+extern "C" int32_t adi_initpinmux(void);
+
+int main(void)
+{
+    SENSOR_RESULT eSensorResult;
+
+    CN0357 coSensor;
+    Gas * gas = &coSensor;
+
+    InitSystem();
+
+    eSensorResult = gas->open();
+
+    if(eSensorResult != SENSOR_ERROR_NONE) {
+        PRINT_SENSOR_ERROR(DEBUG_MESSAGE, eSensorResult);
+        return (1u);
+    }
+
+#if (ADI_APP_USE_BLUETOOTH == 1u)
+
+    if(GasSensorBluetoothMode(gas) == 1u) {
+        return (1u);
+    }
+
+#else
+
+    if(GasSensorStandaloneMode(gas) == 1u) {
+        return (1u);
+    }
+
+#endif
+
+    eSensorResult = gas->close();
+
+    if(eSensorResult != SENSOR_ERROR_NONE) {
+        PRINT_SENSOR_ERROR(DEBUG_MESSAGE, eSensorResult);
+        return (1u);
+    }
+
+    return (0u);
+}
+
+/*!
+ * @brief      Hardware Initialization
+ *
+ * @details    Setups up the power subsystem and the pinmuxing.
+ *
+ */
+static void InitSystem(void)
+{
+    ADI_PWR_RESULT  ePwr;
+
+    /* Explicitly disable the watchdog timer */
+    *pREG_WDT0_CTL = 0x0u;
+
+    /* Pinmux */
+    adi_initpinmux();
+
+    /* Initialize clocks */
+    ePwr = adi_pwr_Init();
+    DEBUG_RESULT("Error initializing the power service.\r\n", ePwr, ADI_PWR_SUCCESS);
+
+    ePwr = adi_pwr_SetClockDivider(ADI_CLOCK_HCLK, 1u);
+    DEBUG_RESULT("Error configuring the core clock.\r\n", ePwr, ADI_PWR_SUCCESS);
+
+    ePwr = adi_pwr_SetClockDivider(ADI_CLOCK_PCLK, 1u);
+    DEBUG_RESULT("Error configuring the peripheral clock.\r\n", ePwr, ADI_PWR_SUCCESS);
+
+    common_Init();
+
+    DEBUG_MESSAGE("Starting CN0357 Demo application\r\n");
+
+}
+
+#if ADI_APP_USE_BLUETOOTH == 1u
+
+/*!
+ * @brief      Bluetooth C0 gas demo.
+ *
+ * @details    Bluetooth mode sends the CO ppm values to the Android or
+ *             ios application.
+ */
+uint8_t GasSensorBluetoothMode(Gas *gas)
+{
+    ADI_BLER_RESULT eResult;
+    float           fConcentration;
+    SENSOR_RESULT   eSensorResult;
+
+    InitBluetooth();
+
+    while(1u) {
+        /* Dispatch events for two seconds - they will arrive in the application callback */
+        eResult = adi_ble_DispatchEvents(ADI_APP_DISPATCH_TIMEOUT);
+        DEBUG_RESULT("Error dispatching events to the callback.\r\n", eResult, ADI_BLER_SUCCESS);
+
+        /* If connected, send data */
+        if(gbConnected == true) {
+            adi_ble_GetConnectionInfo(&sConnInfo);
+
+            if(*reg_flag == 0)	{
+            	eResult = adi_radio_DE_SendData(sConnInfo.nConnHandle,
+            			sizeof(reg_pkt),(uint8_t*)&reg_pkt);
```

Comment:
nitpick: sizeof does not need brackets in this case.

Reply:
For readability I will keep the brackets.